server/src/delta_cache.rs

```diff
@@ -0,0 +1,286 @@
+use std::collections::VecDeque;
+use unleash_types::client_features::{DeltaEvent, ClientFeature, Segment};
+
+
+
+#[derive(Debug, Clone)]
+pub struct DeltaHydrationEvent {
+    event_id: i32,
+    features: Vec<ClientFeature>,
+    segments: Vec<Segment>,
+}
+
+
+#[derive(Debug, Clone)]
+pub struct DeltaCache {
+    max_length: usize,
+    events: VecDeque<DeltaEvent>,
+    hydration_event: DeltaHydrationEvent,
+}
+
+trait GetEventId {
```

Comment:
A trait feels like massive overkill here. Can't we put this method on the DeltaEvent impl?

Reply:
You could also consider just making the fields of the structs public, so you can go `struct.field` instead of needing a `get_field()` method. After all, this isn't Java :)

Reply:
we did the trait because the type was in another lib and was and an enum. Each variant of the enum has the same eventId. This was a workaround to do all of the work in one repo.

Reply:
Can we rather just move this method into the `impl` in the other crate? Events Ids are pretty core to the concept of events, I see no reason to scatter the logic over multiple points if we don't need to

Reply:
Yes, we can move it. We actually did implement it in rs-types and then moved the code to edge as trait so we would not need to create new version. But we can go back easily.

Reply:
Moved